### PR TITLE
[FLINK-13850] [Connectors / FileSystem] Refactor part file configuration into a single method

### PR DESCRIPTION
**Pull request by eskabetxe**
## What is the purpose of the change
- Refactor part file configuration into a single method
- add javadoc for part file configuration usage

## Verifying this change
This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: ( no)
  - The S3 file system connector: ( no )

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (JavaDocs )

### CODE REVIEW

**Comment by eskabetxe** [comment]
@flinkbot attention @gyfora

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java [hunk 1]
```diff
@@ -34,19 +34,34 @@
 
 	private final String partSuffix;
 
+	/**
+	 * Initiates the {@code PartFileConfig} with default values [{@value DEFAULT_PART_PREFIX}, {@value DEFAULT_PART_SUFFIX}].
+	 */
 	PartFileConfig() {
```
Comment (gyfora) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: The PartFileConfig class itself, and all constructors and methods should be public so people can create it.
I think we should also make the fields non-final and add setters (or a builder).

Ideally the user will create a new PartFileConfig with the defaults and override some parameters.

Comment (eskabetxe) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: created a builder and make public the constructor

**Comment by kl0u** [comment]
Hi @eskabetxe ! I think it is worth having a look also at the related discussion in the JIRA https://issues.apache.org/jira/browse/FLINK-13850. 

Given that you are working on the issue, it is also worth telling us what you think on the topic.

**Comment by eskabetxe** [comment]
@kl0u I cant assign issues to me...

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java [hunk 2]
```diff
@@ -18,36 +18,81 @@
...
 /**
  * Part file name configuration.
  * This allow to define a prefix and a suffix to the part file name.
  */
-class PartFileConfig {
-
-	public static final String DEFAULT_PART_PREFIX = "part";
-
-	public static final String DEFAULT_PART_SUFFIX = "";
+public class PartFileConfig implements Serializable {
```
Comment (gyfora) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: Based on the discussion in https://issues.apache.org/jira/browse/FLINK-13850, let's rename this config class to OutputFileConfig

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java [hunk 3]
```diff
@@ -18,36 +18,81 @@
...
+
+		private static final String DEFAULT_PART_PREFIX = "part";
+
+		private static final String DEFAULT_PART_SUFFIX = "";
+
+		private String partPrefix;
+
+		private String partSuffix;
+
+		public PartFileConfigBuilder() {
```
Comment (gyfora) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: The constructor should be private

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java [hunk 4]
```diff
@@ -303,36 +295,30 @@ private BulkFormatBuilder(
...
+			return new BulkFormatBuilder<>(basePath, writerFactory, bucketAssigner, bucketCheckInterval, Preconditions.checkNotNull(factory), partFileConfig);
 	}
 
-		public StreamingFileSink.BulkFormatBuilder<IN, BucketID> withPartFilePrefix(final String partPrefix) {
-			return new BulkFormatBuilder<>(basePath, writerFactory, bucketAssigner, bucketCheckInterval, bucketFactory, partPrefix, partFileSuffix);
-		}
-
-		public StreamingFileSink.BulkFormatBuilder<IN, BucketID> withPartFileSuffix(final String partSuffix) {
-			return new BulkFormatBuilder<>(basePath, writerFactory, bucketAssigner, bucketCheckInterval, bucketFactory, partFilePrefix, partSuffix);
+		public StreamingFileSink.BulkFormatBuilder<IN, BucketID> withPartFileConfig(final PartFileConfig partFileConfig) {
```
Comment (gyfora) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java]: Let's also rename this to withOutputFileConfig accordingly

**Comment by gyfora** [comment]
Would be also great to add some minimal docs to 

https://github.com/apache/flink/blob/master/docs/dev/connectors/streamfile_sink.md

we have made some effort to have good docs for this connector, lets try to keep it up to date with new features :)

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java (hunk 2 above)
Comment (eskabetxe) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: done

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java (hunk 3 above)
Comment (eskabetxe) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/PartFileConfig.java]: done

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java (hunk 4 above)
Comment (eskabetxe) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java]: done

**Comment by eskabetxe** [comment]
@gyfora added a very basic doc..

**Comment by gyfora** [comment]
The changes look good, if you could please squash them together into a single commit I will merge it 👍

**Comment by eskabetxe** [comment]
@gyfora rebased and squashed in a single commit

**Comment by gyfora** [comment]
Thank you for your contribution and sorry for the horribly slow process from my part.

**Pull request closed without merging**